play-validations/memory-footprint/src/main/java/com/google/wear/watchface/dfx/memory/ArscResource.java

```diff
@@ -0,0 +1,86 @@
+package com.google.wear.watchface.dfx.memory;
+
+import java.nio.file.Path;
+
+/**
+ * Represents a resource, as defined in the resources.arsc file within an Android package.
+ */
+public class ArscResource {
+    // Resource type, for example "raw", "asset", "drawable" etc.
+    private final String resourceType;
+
+    // Resource name, for example "watchface" for res/raw/watchface.xml.
+    private final String resourceName;
+
+    // File extension of the resource, for example "xml" for res/raw/watchface.xml
+    private final String extension;
+
+    // Path in the package. This is the obfuscated path to the actual data, where obfuscation has
+    // been used, for example "res/raw/watchface.xml" may point to something like "res/li.xml".
+    private final Path filePath;
+
+    // The resource data itself.
+    private final byte[] data;
+
+    public ArscResource(
+            String resourceType,
+            String resourceName,
+            String extension,
+            Path filePath,
+            byte[] data
+    ) {
+        this.resourceType = resourceType;
+        this.resourceName = resourceName;
+        this.extension = extension;
+        this.filePath = filePath;
+        this.data = data;
+    }
+
+    public String getResourceType() {
+        return resourceType;
+    }
+
+    public String getResourceName() {
+        return resourceName;
+    }
+
+    public String getExtension() {
+        return extension;
+    }
+
+    public Path getFilePath() {
+        return filePath;
+    }
+
+    public byte[] getData() {
+        return data;
+    }
+
+    // TODO: This should be improved to parse res/xml/watch_face_info.xml where present, so as not
+    // to assume all XML files in the res/raw directory are watch face XML files.
+    public Boolean isWatchFaceXml() {
+        return extension.equals("xml") && resourceType.equals("raw");
+    }
+
+    public Boolean isDrawable() {
+        return resourceType.equals("drawable");
+    }
+
+    public Boolean isFont() {
+        return resourceType.equals("font");
+    }
+
+    public Boolean isAsset() {
+        return resourceType.equals("asset");
+    }
+
+    public Boolean isRaw() {
+        return resourceType.equals("raw");
+    }
+
+    public Boolean isWatchFaceInfoFile() {
+        return resourceType.equals("xml") &&
```

Comment:
Reverse all these statements so that you don't potentially get a null pointer exception
```
"xml".equals(resourceType) && ...
```

Reply:
After tests, should we switch to Kotlin? Helps with NPEs.

Reply:
+1 if GitHub is to be the source of truth going forward